engine/src/main/java/org/hibernate/search/engine/search/aggregation/dsl/RangeAggregationOptionsStep.java

```diff
@@ -15,6 +20,44 @@
  * @param <A> The type of result for this aggregation.
  */
 public interface RangeAggregationOptionsStep<S extends RangeAggregationOptionsStep<?, F, A>, F, A>
-		extends AggregationFinalStep<A> {
+	extends AggregationFinalStep<A> {
 
+	/**
+	 * Start describing the behavior of this sort when a document do have
+	 * multi values for the targeted field.
+	 *
+	 * @param mode The renge.
+	 * @return The next step.
+	 */
+	S mode(MultiValue mode);
```

Comment:
1. Exposing a "mode" is not necessary to implement filters on aggregations. Those are two unrelated problems, which should be addressed in two separate tickets and pull requests.
2. Exposing a "mode" for aggregations is not a good idea, as I explained multiple times. It cannot be supported for Elasticsearch, in particular, so if we ever expose it, it will have to be in Lucene-specific extensions to the DSL.
3. Correctly implementing aggregations on multi-valued fields is indeed necessary, but not necessary for this PR. I will keep your code around and submit another PR later, with tests.